docs/source/reference/storage.rst

```diff
@@ -1,201 +1,131 @@
-      # this approach can also be used to create a shared filesystem across workers.
-      # See examples/storage/pingpong.yaml for an example.
-      /outputs-mount:
-        name: my-output-bucket
-        mode: MOUNT
-
-      # *** Uploading multiple files to the same Storage object ***
-      #
-      # The source field in a storage object can also be a list of local paths.
-      # This is useful when multiple files or directories need to be uploaded to the
-      # same bucket.
-      #
-      # Note: The basenames of each path in the source list are copied recursively
-      # to the root of the bucket. Thus, if the source list contains a directory,
-      # the entire directory is copied to the root of the bucket. For instance,
-      # in this example, the contents of ~/datasets are copied to
-      # s3://sky-multisource-storage/datasets/. ~/mydir/myfile.txt will appear
-      # at s3://sky-multisource-storage/myfile.txt.
-      /datasets-multisource-storage:
-        name: sky-multisource-storage # Make sure this name is unique or you own this bucket
-        source: [~/mydir/myfile.txt, ~/datasets]
-
-
-    run: |
-      pwd
-      ls -la /
-
-    # Remember to run `sky storage ls` and `sky storage delete` to delete the
-    # created storage objects!
+        To create an empty bucket, specify only the ``name`` and omit the ``source``.
 
-.. note::
-    Stopping a running cluster will cause any Storage mounted with :code:`MOUNT`
-    mode to be unmounted. These mounts will not be re-mounted on running
-    :code:`sky start`, or even :code:`sky exec`. Please run :code:`sky launch`
-    again on the same cluster to ensure :code:`MOUNT` mode Storages are mounted
-    again.
+        .. code-block:: yaml
+
+          # Create an empty gcs bucket
+          file_mounts:
+            /mydata:
+                name: my-sky-bucket
+                store: gcs   # Optional - either of s3, gcs, r2, ibm
+
+        SkyPilot will create an empty GCS bucket called ``my-sky-bucket`` and mount it at ``/mydata``.
+        This empty bucket can be used to write checkpoints, logs or other outputs directly to the cloud.
+
+You can find more detailed usage examples in `storage_demo.yaml <https://github.com/skypilot-org/skypilot/blob/master/examples/storage_demo.yaml>`_.
+
+.. _storage-mounting-modes:
+
+Mounting modes
+--------------
+
+A cloud storage can used in either :code:`MOUNT` mode or :code:`COPY` mode.
+
+1. **MOUNT** mode: The object store is directly "mounted" to the remote VM. I.e., files are streamed when accessed by the task and all writes are replicated to remote bucket (and any other VMs mounting the same bucket). This is the default mode.
+2. **COPY** mode: The files are pre-fetched and cached on the local disk. Writing to object stores is not supported in this mode.
+
+.. Source for the image: https://docs.google.com/drawings/d/1MPdVd2TFgAFOYSk6R6E903v1_C0LHmVU-ChIVwdX9A8/edit?usp=sharing
+
+.. image:: ../images/sky-storage-modes.png
```

Comment:
This is excellent! Very clear. 

Some first impression confusion
- `my_data` -> `mydata` to be consistent with snippet above
- in the COPY mode box, the `open(..)` arrow should be two-directional to reflect writes can affect Local Path